### ISSUE TITLE
[a11y] Consider adding empty alt attribute on <img when none is provided

### ISSUE THREAD

**Issue by gmajoulet** [ampproject/amphtml#34278]
On regular AMP pages, when publishers don't provide an `alt` attribute on their `amp-img`, the underlying `img` doesn't get any `alt`.
Most screenreaders will fallback on reading the `href` of the `<img` tag, which might or might not be useful.

For AMP Stories specifically, we decided that we'd rather have screenreaders remain silent than reading the image's source URL, and add an empty `alt` attribute to do so.

https://github.com/ampproject/amphtml/blob/06edda1fec83bb9a8eadb6f3633b042001ccc672/extensions/amp-story/1.0/amp-story-page.js#L1842-L1859

The Stories working group doesn't have enough content to make a decision for regular AMP pages, but we'd like to open the conversation and bring this to @ampproject/wg-components attention, as this might be a great accessibility improvement.

cc @ampproject/wg-stories FYI

**Comment by kristoferbaxter** [comment]
@caroqliu This seems to be a decent default behaviour, should we apply this to `amp-img` globally?

**Comment by caroqliu** [comment]
I can see why silencing screen readers for unlabelled image content is sensible for Stories format since there are already known issues with verbosity, in addition to a reasonable assumption that images in a Story or more likely to be [decorative](w3.org/WAI/tutorials/images/decorative/) than those on regular AMP pages.

It seems like appending `alt=""` can be preferred to the alternative reading of `src` because like @gmajoulet said, verbalizing `src` can be both unhelpful and verbose. And to be clear that is the guidance for decorative images. 

But for AMP for websites we don't often know what the purpose of the image is, so I'm not confident that treating all non-labelled `amp-img` as decorative (and therefore silencing reading) is an appropriate default either. Talked with @PeterSaathoff-Harshfield about this some and covered a few counterpoints/considerations: 
- If there is an image that contains a lot of text, and surrounding elements assume the user has context from what the image conveys, it can be more disorienting to have had the screen reader skip the image entirely than to have some verbalization (even if it's the `src`). While reading the `src` can be verbose/annoying, it is still expected to effectively convey "there's content I'm missing here." 
- Given the above, it can be more helpful to add `alt="Unlabelled image"` instead of `alt=""` to convey that there is an image there but we can't help discern the content. If we go this route, we should also be careful to follow i18n considerations to make sure "Unlabelled image" is correctly localized based on the document language.
- Even then, we may not be able to assume that "Unlabelled image" is more often better than reading "src" outright, since sometimes sources can actually contain descriptive file names that may be more helpful to have than "Unlabelled", i.e. "puppy.jpg".

Maybe what we can do instead is get the file name substring (based on the last `/` character in `src`) and provide default `alt="Unlabelled image with filename ${substring}"`? What do folks think?

cc @davidstrauss in case you have insights here as well?

**Comment by PeterSaathoff-Harshfield** [comment]
Another consideration is that screen readers already announce an element type for images, with some saying "image" and some saying "graphic", at least for English, so we want to avoid  providing a redundant element type in the image alt tag. The unlabeled label should only say "unlabeled", otherwise screen readers will say "unlabeled image, image". Similarly if we try to strip off the beginning of the URL before a filename, I'd lean toward making the alt value only be that filename and not include phrasing like "unlabeled image with filename" because users will already understand that a) it's an image, b) a good label was not provided, and c) this is probably a filename.

**Comment by caroqliu** [comment]
> Another consideration is that screen readers already announce an element type for images, with some saying "image" and some saying "graphic", at least for English, so we want to avoid providing a redundant element type in the image alt tag. The unlabeled label should only say "unlabeled", otherwise screen readers will say "unlabeled image, image". Similarly if we try to strip off the beginning of the URL before a filename, I'd lean toward making the alt value only be that filename and not include phrasing like "unlabeled image with filename" because users will already understand that a) it's an image, b) a good label was not provided, and c) this is probably a filename.

You mean something like "unlabelled filename" could make a good default `alt`?

**Comment by kristoferbaxter** [comment]
Like these suggestions as we can also include this as the default for SSR of these components in the Optimizer and on Caches.

This means we should not need to ship the logic for the client in the module build.

**Comment by PeterSaathoff-Harshfield** [comment]
> What about `"unlabelled <stripped filename>"` as a combined option that's more simplified than `"unlabelled image with <filename>"`?

Sounds good to me.

**Comment by gmajoulet** [comment]
Is there any precedent to solve the i18n considerations here?

**Comment by kristoferbaxter** [comment]
There is not a precedence for the i18n consideration, but we should cross the bridge.

**Comment by gmajoulet** [comment]
Happy to help with the context the Stories team has on the localization service/process. Should we add this into some team's roadmap? It likely will be a bit more than just a few lines of code. cc @nainar

**Comment by newmuis** [comment]
+1 to additional localization support being needed to resolve this.  There are a few unsolved challenges there including:

1. *No placeholder support.*  A string like the `unlabelled <stripped filename>` suggestion from @caroqliu above is simple in English; it's simply concatenate the filename after `unlabelled`.  But the order and placement will not be the same in every language.  Most mature localization frameworks use placeholders to solve this problem; something akin to `unlabelled %filename`, and programmatically replace `%filename` with the filename.  We do not currently have the ability to do this, although one could certainly build it.
2. *Serving and bundle size.*  Currently, all variants of the string are served to all users, which can be wasteful.  Ideally, if I'm viewing a page in German, I would only get the German strings.  We don't have a way to do this currently. I had some preliminary conversations with @rsimha quite some time ago, but we ended up punting on the problem for the time being; perhaps it's time to revisit.
3. *Strings translations don't happen automatically.*  Until now, string translation has been a very manual process that I've been doing on behalf of the stories working group.  We have plans in motion to get that resolved, but they have not landed yet.
